nuget/lib/dependabot/nuget/file_updater/project_file_declaration_finder.rb

```diff
@@ -95,6 +111,80 @@ def declaring_file
 
           raise "No file found with name #{filename}!"
         end
+
+        def sdk_import_strings
+          sdk_strings(SDK_IMPORT_REGEX, "Import", "Sdk", "Version")
+        end
+
+        # rubocop:disable Metrics/PerceivedComplexity
+        def sdk_project_strings
+          dep_name = dependency_name&.downcase
+          dep_version = declaring_requirement.fetch(:requirement)
+          strings = []
+          declaring_file.content.scan(SDK_PROJECT_REGEX).each do |string|
+            xml = string
+            xml += "</Project>" unless string.end_with?("/>")
+            node = Nokogiri::XML(xml)
+            node.remove_namespaces!
+            element = node.at_xpath("/Project")
+            next unless element
```

Comment:
There's some duplication between this and [the block in `#sdk_strings`](https://github.com/dependabot/dependabot-core/pull/2849/files#diff-fa02b4ba56ca051e40b58c9152d7796a54f9f2f017a304fc911a7edb41aee838R164-R168), could it be worth extracting a helper method?

Reply:
Will have a look

Reply:
This has been refactored, with several ignore-complexity comments removed.